src/vs/platform/externalTerminal/node/externalTerminalService.ts

```diff
@@ -369,3 +375,152 @@ function quote(args: string[]): string {
 	}
 	return r;
 }
+
+namespace KittyTerminalService {
+	export interface KittyProcess {
+		cmdline: string[];
+		cwd: string;
+		pid: number;
+	}
+
+	export interface KittyWindow {
+		id: number;
+		title: string;
+		at_prompt: boolean;
+		foreground_processes: KittyProcess[];
+	}
+
+	export interface KittyTab {
+		windows: KittyWindow[];
+	}
+
+	export interface KittyInstance {
+		tabs: KittyTab[];
+	}
+}
+
+class KittyTerminalService {
+	static readonly KITTY_SOCKET = '/tmp/vscode-kitty.sock';
+	static readonly INVALID_WINDOW_ID = -1;
+
+	static async ensureKittyRunning() {
+		if (!existsSync(this.KITTY_SOCKET) || !statSync(this.KITTY_SOCKET).isSocket()) {
+			const child = cp.spawn('kitty', [
+				'-o', 'allow_remote_control=yes',
+				`--listen-on=unix:${this.KITTY_SOCKET}`
+			], {
+				detached: true,
+				stdio: 'ignore'
+			});
+			child.on('error', () => {
+				throw (new Error(nls.localize('mac.terminal.kitty.not.found', "Kitty terminal is not found.")));
+			});
+			child.unref(); // Fully detach the process
+
+			// Wait for the socket to be created
+			for (let i = 0; i < 10; i++) { // Retry for ~5 seconds
+				await delay(500);
+				if (existsSync(this.KITTY_SOCKET) && statSync(this.KITTY_SOCKET).isSocket()) {
+					return;
+				}
+			}
+			throw (new Error(nls.localize('mac.terminal.kitty.not.running', "Kitty terminal is not running")));
+		}
+	}
+
+	private static getInstances(): KittyTerminalService.KittyInstance[] {
+		try {
+			const result = cp.spawnSync('kitten', ['@', '--to', `unix:${this.KITTY_SOCKET}`, 'ls'], {
+				stdio: 'pipe',
+			});
+			if (result.status !== 0) {
+				const errorOutput = result.stderr.toString().trim() || 'Unknown error';
+				throw new Error(nls.localize('mac.terminal.kitty.list.failed', "Failed to list kitty windows: {0}", errorOutput));
+			}
+			const output = result.stdout.toString();
+			const instances = JSON.parse(output);
+			return instances;
```

Comment:
Can we validate the output properly in an `is*` function, see this for example: https://github.com/microsoft/vscode/blob/96e35391ab391e7ecd13050e02785c12905a6f15/src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/tools/monitoring/outputMonitor.ts#L405-L414